src/cairoWriter/writers/sourceUnitWriter.ts

```diff
@@ -41,21 +42,60 @@ export class SourceUnitWriter extends CairoASTNodeWriter {
       writer.write(v),
     );
 
-    const functions = node.vFunctions.map((v) => writer.write(v));
+    const [importFunctions, generatedFunctions, functions] =
+      node.vFunctions.length === 0
+        ? [[], [], []]
+        : node.vFunctions.reduce(
+            ([importFunctions, generatedFunctions, functions], funcDef) =>
+              funcDef instanceof CairoImportFunctionDefinition
+                ? [[funcDef, ...importFunctions], generatedFunctions, functions]
+                : funcDef instanceof CairoGeneratedFunctionDefinition
+                ? [importFunctions, [funcDef, ...generatedFunctions], functions]
+                : [importFunctions, generatedFunctions, [funcDef, ...functions]],
+            [
+              new Array<CairoImportFunctionDefinition>(),
+              new Array<CairoGeneratedFunctionDefinition>(),
+              new Array<FunctionDefinition>(),
+            ],
+          );
+
+    const writtenImportFuncs =
+      importFunctions.length === 0
+        ? ''
+        : getGroupedImports(
+            importFunctions
+              .sort((funcA, funcB) =>
+                `${funcA.path}.${funcA.name}`.localeCompare(`${funcB.path}.${funcB.name}`),
+              )
+              .filter((func, index, importFuncs) => func.name !== importFuncs[index - 1]?.name),
+          ).reduce((writtenImports, importFunc) => `${writtenImports}\n${importFunc}`);
```

Comment:
It's the same thing here, but you'll need to provide an initial value for the reduce method. It would look like this:
```suggestion
    const writtenImportFuncs = getGroupedImports(
          importFunctions
            .sort((funcA, funcB) =>
              `${funcA.path}.${funcA.name}`.localeCompare(`${funcB.path}.${funcB.name}`),
            )
            .filter((func, index, importFuncs) => func.name !== importFuncs[index - 1]?.name),
        ).reduce((writtenImports, importFunc) => `${writtenImports}\n${importFunc}`, '');
```